compiler/semexprs.nim

```diff
@@ -1094,7 +1094,7 @@ proc builtinFieldAccess(c: PContext, n: PNode, flags: TExprFlags): PNode =
     of tyObject, tyTuple:
       if ty.n != nil and ty.n.kind == nkRecList:
         for field in ty.n:
-          if field.sym.name == i:
+          if field.kind == nkSym and field.sym.name == i:
```

Comment:
I will accept this, but the original code is all wrong. It needs to iterate over the 'ty.n' properly.